### DISCUSSION TITLE
Background on User Profile

### DISCUSSION THREAD

**Discussion by jvwong** [PathwayCommons/hyper-recent#10] [Ideas]
### General

Our target users are biological and medical researchers, that is, “domain experts”. Their primary need is to stay informed about advancements in their research field. They may also be curious about notable efforts on the margins or outside their expertise, because it could inspire their own work or due the amount of attention, but these are secondary. 

These researchers are very familiar with PubMed and use it regularly to search for articles. The target users are interested in collecting information related to a topic, and so would typically use PubMed by submitting “informational queries”. This is unlike “navigational queries” used to locate a specific record or set. It is important to note that like most PubMed visitors, our target users have no background or specialized training with databases, knowledge representation or search. 

Given users’ needs, when they have a set of search criteria, our researchers would expect to receive a set of articles ordered by date, akin to a feed. It would not necessarily be problematic for a system to return zero new results (10% of PubMed queries). Domain experts are also quite tolerant of noise in sets of informational search hits, because they are considerate of the qualitative difference between content provided by bibliographic indexes and, say, Google. Indeed, their expectations are unlike those of Google or Amazon where one expects a ranked list of possibly millions of results. 

Our users are accustomed to submitting queries consisting of a few terms. Like PubMed and search engines, the overwhelming majority (>90%) of users submit single queries in a brief session; with only a minority interested in subsequently refining an initial search. Target users are error-prone, and a significant proportion of input search terms are misspelled (10-20%) in line with observations from PubMed. On average, target users enter three (3) terms in a typical query, with the majority (70%) entering up to four (4) terms to define the content they see. Being non-experts in search and having informational needs, they will not entertain the notion of annotating terms with specialized tags or syntax (e.g. operators) in queries and typically not interested in advanced search functions. All of this notwithstanding, it should be noted that a large proportion of PubMed queries in general include an author name (35%). 

### Individual Users

Rather than define prototypical users _a priori_ and derive hypothetical queries, we do the inverse: randomly sample queries from 1 day of PubMed logs ([Wilbur WJ, Kim W, Xie N. 2006. Spelling correction in the PubMed search engine. Information Retrieval 9](https://link.springer.com/article/10.1007/s10791-006-9002-8) accessible at https://ftp.ncbi.nlm.nih.gov/pub/wilbur/DAYSLOG/ and attempt to infer their intentions. Below in Table I, I randomly sample 1000 lines of the PubMed log using the command: `$ shuf -n 10 pubmed-queries.txt > sample-pubmed-queries.txt` and have (arbitrarily) selected 30 queries of interest.

**Table 1. Informational query samples**

<table>
  <tr>
   <td><strong>Query</strong>
   </td>
   <td><strong>Types</strong>
   </td>
   <td><strong>Comments</strong>
   </td>
  </tr>
  <tr>
   <td><code>Michael Elowitz</code>
   </td>
   <td>author 
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>delta opioid receptor</code>
   </td>
   <td>gene family
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>propofol, sizure</code>
   </td>
   <td>medication; neurological event
   </td>
   <td>misspelling (<code>seizure</code>)
   </td>
  </tr>
  <tr>
   <td><code>top 10 deseases for males</code>
   </td>
   <td>sex-based disease
   </td>
   <td>misspelling (<code>diseases</code>)
   </td>
  </tr>
  <tr>
   <td><code>Christopher Barrett</code>
   </td>
   <td>author
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>fetal hemoglobin congeintal heart disease</code>
   </td>
   <td>developmental protein; disease
   </td>
   <td>misspelling (<code>congenital</code>)
   </td>
  </tr>
  <tr>
   <td><code>hypoxia sex hormones</code>
   </td>
   <td>state; protein family
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>rhabdomyolysis and labetalol</code>
   </td>
   <td>pathological process; medication
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>swi promoter escape</code>
   </td>
   <td>protein complex; molecular process
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>Kawasaki</code>
   </td>
   <td>disease 
   </td>
   <td>Partial “Kawasaki disease”
   </td>
  </tr>
  <tr>
   <td><code>jessell t </code>
   </td>
   <td>author
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>Thymic selection</code>
   </td>
   <td>cellular process
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>gene silencing plant</code>
   </td>
   <td>molecular process; taxon
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>ampk</code>
   </td>
   <td>protein kinase
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>reproducibility</code>
   </td>
   <td>scientific integrity
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>muscle myogenesis</code>
   </td>
   <td>developmental process
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>Bra1 brain protein</code>
   </td>
   <td>protein; organ
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>psychometric properties hawaii early learning profile</code>
   </td>
   <td>curriculum-based assessment tool
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>rheumatoid arthritis X-rays scoring methods reliability</code>
   </td>
   <td>disease; clinical assessment method
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>cornea and oxidative stress</code>
   </td>
   <td>tissue; stimulus
   </td>
   <td>Not operator
   </td>
  </tr>
  <tr>
   <td><code>Lipossomes cancers Karposi Sarcoma</code>
   </td>
   <td>organelle; disease
   </td>
   <td>misspelling (<code>Liposomes; Kaposi</code>)
   </td>
  </tr>
  <tr>
   <td><code>brteast cancer positive nodes</code>
   </td>
   <td>disease staging
   </td>
   <td>misspelling (<code>breast</code>)
   </td>
  </tr>
  <tr>
   <td><code>tamoxifen MBC</code>
   </td>
   <td>drug; disease stage 
   </td>
   <td>Abbreviation for “Male Breast Cancer”
   </td>
  </tr>
  <tr>
   <td><code>cyclosporin, expoliative dermatitis</code>
   </td>
   <td>drug; pathological condition
   </td>
   <td>misspelling (<code>exfoliative</code>)
   </td>
  </tr>
  <tr>
   <td><code>pancreatitis c reactive peptide</code>
   </td>
   <td>disease; 
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>depression and fluoxetine</code>
   </td>
   <td>disease state; antidepressant
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>Hac1 yeast mammanlian</code>
   </td>
   <td>gene; organism
   </td>
   <td>misspelling (<code>mammalian</code>)
   </td>
  </tr>
  <tr>
   <td><code>vegf and prostate</code>
   </td>
   <td>gene; organ
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td><code>cutaneous effects of topamax</code>
   </td>
   <td>side-effect of medication
   </td>
   <td>
   </td>
  </tr>
</table>
 Excluded queries with: Nothing (empty); tags; local IDs; non-english; citations; garbage

## Background

**Opposing retrieval models**

_exact-match_ (Boolean)

* The query specifies precise retrieval criteria
* Every document either matches or fails to match the query
* The result is a set of documents

_best-match_ (Information retrieval)

* The query describes a good or “best” matching document
* The result is a ranked list of documents
* The result may include an estimate of quality

**Notes on library science tradition vs information & computer paradigm**

* Information and computer paradigm
    * emphasize “query transformation” which transforms a users verbal query into a set of documents or bibliographic records
    * reify “relevance” towards the construction of a perfect system
* Library science tradition
    * “selection power” is a user’s ability to make relevant distinctions during a search
    * There is a need for human labor to create this selection power
* Warner J has stated that boolean searches have the aim of increasing the “selection power” of users because of the _iterative_ nature of the search and users’ control over the process
* The quality of search hits, as judged by a user, may evolve over time in response to each item of new information, such as prior queries

**Criticisms of Boolean **

1. Requires user training in arcane query syntax, knowledge organization
2. Too restrictive (AND)
3. Too inclusive/noisy (OR)
4.  (2) and (3) result in search failure or overload
5. No priority or weighting of terms
6. Boolean logic requires that terminology for a user’s query and the document index align
7. Boolean logic precludes document ranking

Only (1) seems valid after refuting these criticisms.

**On IR and the popularity of Google**

_There is no question that systems such as Google, based on a kind of scoring system, are easy to use and highly popular. But much of the popularity of contemporary search engines may also be attributed to the easy pickings afforded by the first generation of Internet full-text based systems (owing to the cheap cost of digital storage capacity after 1990): no doubt it is good to have all text on the web indexed and made searchable—and often with free access. However, when the easy pickings have been utilized, more complex strategies (and more humanistic approaches) may be needed to make further progress._

**PubMed Log analysis**

* 24 hours of log analysis
    * Total: 2 996 301 queries issued by 627 455 users
    * Type
        * informational (no search field tag): 2 585 183 (89%)
        * navigation (tag): 11%
    * Queries
        * Median 3 terms per query
        * Majority (90%) had single session during the day
            * Largely 1 query per session

## References 

1. Hjørland, B. Classical databases and knowledge organization: A case for boolean retrieval and human decision-making during searches. J Assoc Inf Sci Tech 66, 
2. Herskovic, J. R., Tanaka, L. Y., Hersh, W. & Bernstam, E. V. A Day in the Life of PubMed: Analysis of a Typical Day’s Query Log. J Am Med Inform Assn 14, 212–220 (2007).
3. Fiorini, N., Leaman, R., Lipman, D. J. & Lu, Z. How user intelligence is improving PubMed. Nat Biotechnol 36, 937–945 (2018).

**Comment by maxkfranz** [comment]
One of the most challenging issues coming out of those sample queries is misspellings.  In order to accommodate frequent misspellings, you need to address domain-specific terms that aren't English dictionary words (e.g. tnf).  For instance, you could spell check only English words -- then you need to classify general English misspellings from non-English words like 'tnf'.
